docs/public/robots.txt

```diff
@@ -1,2 +1,3 @@
 User-agent: *
 Allow: /
+Disallow: /x/react-scheduler/
```

Comment:
Do you know if I should keep the `/x` here or not?
I guess I should

Reply:
I assume yes, since all the links in this repo start with `/x/`.
However, I fear that this could cause issues with caching this disallow and then requiring some time for the removal to take effect... 🤔 
Do we strictly need this to not be indexed? 🤷 

WDYT @mui/code-infra, do you see any potential problems with this?